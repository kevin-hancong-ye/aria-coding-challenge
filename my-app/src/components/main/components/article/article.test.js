import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";

import Article from "./article";

test("renders article component", () => {
	render(<Article />);

	const headingElem = screen.getByRole("heading");
	expect(headingElem).toHaveTextContent("Article");

	const articleOneElem = screen.getByTestId("article-1");
	expect(articleOneElem).toBeInTheDocument();
	expect(articleOneElem).toMatchSnapshot();

	const articleTwoElem = screen.getByTestId("article-2");
	expect(articleTwoElem).toBeInTheDocument();
	expect(articleTwoElem).toMatchSnapshot();
});
