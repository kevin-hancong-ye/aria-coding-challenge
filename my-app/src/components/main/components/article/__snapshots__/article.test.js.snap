// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders banner component 1`] = `
<div
  data-testid="article-1"
  role="article"
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. In quibus doctissimi illi veteres inesse quiddam caeleste et divinum putaverunt. Hoc enim identidem dicitis, non intellegere nos quam dicatis voluptatem. Duo Reges: constructio interrete.
    <b>
      Primum in nostrane potestate est, quid meminerimus?
    </b>
    Si verbum sequimur, primum longius verbum praepositum quam bonum. Non est igitur summum malum dolor. Quis istud possit, inquit, negare?
  </p>
</div>
`;

exports[`renders banner component 2`] = `
<div
  data-testid="article-2"
  role="article"
>
  <p>
    Primum cur ista res digna odio est, nisi quod est turpis? Bonum incolumis acies: misera caecitas.
    <i>
      Bonum liberi: misera orbitas.
    </i>
     Haeret in salebra. Verba tu fingas et ea dicas, quae non sentias? Negat esse eam, inquit, propter se expetendam.
  </p>
  <p>
    Nam et a te perfici istam disputationem volo, nec tua mihi oratio longa videri potest. Habent enim et bene longam et satis litigiosam disputationem. Tanti autem aderant vesicae et torminum morbi, ut nihil ad eorum magnitudinem posset accedere. 
    <i>
      Res enim concurrent contrariae.
    </i>
     Voluptatem cum summum bonum diceret, primum in eo ipso parum vidit, deinde hoc quoque alienum; Non igitur potestis voluptate omnia dirigentes aut tueri aut retinere virtutem. Quaesita enim virtus est, non quae relinqueret naturam, sed quae tueretur. In qua quid est boni praeter summam voluptatem, et eam sempiternam? Ita relinquet duas, de quibus etiam atque etiam consideret. Erat enim Polemonis. Fortemne possumus dicere eundem illum Torquatum? Ab hoc autem quaedam non melius quam veteres, quaedam omnino relicta.
  </p>
</div>
`;
